### PR TITLE
CTSM package organization

### PR DESCRIPTION
**Pull request by tonygardella**
In accordance with issue #2323 , I'm moving over the functions from the PEcAn.FATES package to PEcAn.CTSM so we can start correcting the scripts to work with the latest CTSM versions. This PR will handle naming functions and the base organization of the directory. 

Also, this merges in the latest code from DEVELOP so you'll see all those commits too.

## Review Time Estimate

- [ ] Immediately
- [x] Within one week
- [ ] When possible
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue) 
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the CHANGELOG.md.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by tonygardella** [comment]
@dlebauer can you approve this before #2365. The commit you just did there is redundant to this.

**Comment by tonygardella** [comment]
@dlebauer @serbinsh Can you take a look at this PR?

File: models/ctsm/DESCRIPTION [hunk 1]
```diff
@@ -1,17 +1,40 @@
...
+             person("Tony", "Gardella", role = c("aut"),
+                     email = "tonygard@bu.edu",
+                     comment = c(ORCID = "0000-0003-4380-3412")))
 Description: The Predictive Ecosystem Analyzer (PEcAn) is a scientific
     workflow management tool that is designed to simplify the management of
     model parameterization, execution, and analysis. The goal of PEcAn is to
     streamline the interaction between data and models, and to improve the
     efficacy of scientific investigation. This package provides functions to
-    link the Community Land Model, versions 4.5 and 5, to PEcAn.
+    link the Community Terrestrial System Model, versions 5, to PEcAn. Will
```
Comment (dlebauer) [inline models/ctsm/DESCRIPTION]: one thing we really need to do is figure out how to be consistent about naming ... is CTSM a synonym for CLM5?  which should we use? Can we just say that CLM5 is a synonym for CTSM once, and then use CTSM everywhere else?

Comment (tonygardella) [reply models/ctsm/DESCRIPTION]: Don't think it's a synonym. In issue #2122 we say that CTSM package is a support package for LM5 and FATES.

**Comment by KristinaRiemer** [comment]
@tonygardella this looks good to me.

**Pull request merged**